### PR TITLE
Unify core unit tests

### PR DESCRIPTION
**Pull request by masterleinad**
This pull request unifies the unit tests in `core`.

### CODE REVIEW

File: core/src/Kokkos_HPX.hpp [hunk 1]
```diff
@@ -566,6 +566,11 @@ class TeamPolicyInternal<Kokkos::Experimental::HPX, Properties...>
            team_size_ * m_thread_scratch_size[level];
   }
 
+   inline static int scratch_size_max(int level) {
+    return (level == 0 ? 1024 * 32 :  // Roughly L1 size
+                20 * 1024 * 1024);    // Limit to keep compatibility with CUDA
+  }
```
Comment (masterleinad) [inline core/src/Kokkos_HPX.hpp]: Stolen from `Serial` and `OpenMP`.

File: .travis.yml [hunk 2]
```diff
@@ -88,7 +88,7 @@ script:
           -DKokkos_ENABLE_TESTS=On
           ${CMAKE_BUILD_TYPE:+-DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}} &&
     make VERBOSE=1 -j2 &&
-    travis_wait make test CTEST_OUTPUT_ON_FAILURE=1 &&
+    travis_wait 60 make test CTEST_OUTPUT_ON_FAILURE=1 &&
```
Comment (dalg24) [inline .travis.yml]: See https://github.com/kokkos/kokkos/pull/2450#discussion_r335410475

File: core/src/Kokkos_HPX.hpp (hunk 1 above)
Comment (dalg24) [reply core/src/Kokkos_HPX.hpp]: @msimberg please review this addition.

File: core/unit_test/CMakeLists.txt [hunk 3]
```diff
@@ -24,200 +24,108 @@ KOKKOS_TARGET_INCLUDE_DIRECTORIES(kokkos_gtest PUBLIC ${GTEST_SOURCE_DIR})
...
+      ${dir}/Test${Tag}_ViewMapping_subview.cpp
+      ${dir}/Test${Tag}_ViewOfClass.cpp
+      ${dir}/Test${Tag}_Crs.cpp
+      ${dir}/Test${Tag}_WorkGraph.cpp
+      ${dir}/Test${Tag}_View_64bit.cpp
+      ${dir}/Test${Tag}_ViewResize.cpp
+  )
+endforeach()
+
+IF(Kokkos_ENABLE_SERIAL)
```
Comment (dalg24) [inline core/unit_test/CMakeLists.txt]: Lowercase if for consistency.

File: core/unit_test/CMakeLists.txt [hunk 4]
```diff
@@ -24,200 +24,108 @@ KOKKOS_TARGET_INCLUDE_DIRECTORIES(kokkos_gtest PUBLIC ${GTEST_SOURCE_DIR})
...
-      threads/TestThreads_ViewAPI_d.cpp
-      threads/TestThreads_ViewAPI_e.cpp
-      threads/TestThreads_ViewMapping_a.cpp
-      threads/TestThreads_ViewMapping_b.cpp
-      threads/TestThreads_ViewMapping_subview.cpp
-      threads/TestThreads_ViewOfClass.cpp
-      threads/TestThreads_Crs.cpp
-      threads/TestThreads_WorkGraph.cpp
-      threads/TestThreads_ViewResize.cpp
+    SOURCES ${Threads_SOURCES}
```
Comment (dalg24) [inline core/unit_test/CMakeLists.txt]: @dhollman Does the task test not work with the ptrhreads backend?

File: core/unit_test/CMakeLists.txt [hunk 5]
```diff
@@ -24,200 +24,108 @@ KOKKOS_TARGET_INCLUDE_DIRECTORIES(kokkos_gtest PUBLIC ${GTEST_SOURCE_DIR})
...
-      openmp/TestOpenMP_ViewMapping_b.cpp
-      openmp/TestOpenMP_ViewMapping_subview.cpp
-      openmp/TestOpenMP_ViewOfClass.cpp
-      openmp/TestOpenMP_Crs.cpp
-      openmp/TestOpenMP_WorkGraph.cpp
-      openmp/TestOpenMP_UniqueToken.cpp
-      openmp/TestOpenMP_ViewResize.cpp
+    ${OpenMP_SOURCES}
+    openmp/TestOpenMP_Task.cpp
+    openmp/TestOpenMP_UniqueToken.cpp
```
Comment (dalg24) [inline core/unit_test/CMakeLists.txt]: Why does the `UniqueToken` get special treatment?
Is there a good reason for not running it in serial and with pthreads?

**Review by dalg24 — CHANGES_REQUESTED** [review]
Looks good for the most part.
Did you include the Travis changes by inadvertence or is it actually needed for the tests to pass?  I can't tell from the history.

Thanks for working on this.

**Comment by masterleinad** [comment]
> Did you include the Travis changes by inadvertence or is it actually needed for the tests to pass?

I again needed that for Travis not to timeout.

File: core/unit_test/CMakeLists.txt (hunk 4 above)
Comment (masterleinad) [reply core/unit_test/CMakeLists.txt]: Building the test would currently fail with
```
[ 99%] Building CXX object core/unit_test/CMakeFiles/KokkosCore_UnitTest_Threads.dir/threads/TestThreads_Task.cpp.o
In file included from kokkos/core/src/Kokkos_Core.hpp:94:0,
                 from kokkos/core/unit_test/TestTaskScheduler.hpp:49,
                 from kokkos/core/unit_test/threads/TestThreads_Task.cpp:46:
kokkos/core/src/Kokkos_TaskScheduler.hpp: In instantiation of ‘class Kokkos::BasicTaskScheduler<Kokkos::Threads, Kokkos::Impl::TaskQueue<Kokkos::Threads, Kokkos::HostSpace> >’:
kokkos/core/src/Kokkos_Future.hpp:285:62:   required from ‘class Kokkos::BasicFuture<long int, Kokkos::BasicTaskScheduler<Kokkos::Threads, Kokkos::Impl::TaskQueue<Kokkos::Threads, Kokkos::HostSpace> > >’
kokkos/core/unit_test/TestTaskScheduler.hpp:82:15:   required from ‘struct TestTaskScheduler::TestFib<Kokkos::BasicTaskScheduler<Kokkos::Threads, Kokkos::Impl::TaskQueue<Kokkos::Threads, Kokkos::HostSpace> > >’
kokkos/core/unit_test/TestTaskScheduler_single.hpp:50:47:   required from here
kokkos/core/src/Kokkos_TaskScheduler.hpp:88:67: error: no type named ‘member_type’ in ‘struct Kokkos::Impl::TaskQueueSpecialization<Kokkos::BasicTaskScheduler<Kokkos::Threads, Kokkos::Impl::TaskQueue<Kokkos::Threads, Kokkos::HostSpace> > >’
   using member_type         = typename specialization::member_type;
```

File: core/unit_test/CMakeLists.txt (hunk 5 above)
Comment (masterleinad) [reply core/unit_test/CMakeLists.txt]: Seems to work.

**Comment by masterleinad** [comment]
Verbose test output seems to do the trick here as well, but we have to find something that both works for this pull request and #2450.

**Comment by dalg24** [comment]
@masterleinad Please rebase this onto `develop`

**Comment by masterleinad** [comment]
Done.

File: core/unit_test/CMakeLists.txt (hunk 4 above)
Comment (dhollman) [reply core/unit_test/CMakeLists.txt]: Tasking is not implemented for the pthreads backend AFAIK

**Review by dalg24 — APPROVED** [review]

**Pull request merged**